discord/file.py

```diff
@@ -126,6 +177,24 @@ def filename(self) -> str:
         """
         return 'SPOILER_' + self._filename if self.spoiler else self._filename
 
+    @property
+    def waveform(self) -> list[int]:
+        """List[:class:`int`]: The waveform data for the voice message.
+
+        .. note::
+            If a waveform was not given, it will be generated
+
+            Only supports generating the waveform for Opus format files, other files will be given a random waveform
+
+        .. versionadded:: 2.7"""
+        if self._waveform is None:
+            try:
+                self._waveform = self.generate_waveform()
```

Comment:
Properties should be inexpensive to access. I would move this generation into the init if a waveform is not provided. Doing it on a property access seems rather silly.